pkg/operator/operator.go

```diff
@@ -269,31 +274,82 @@ func (o *Operator) setupAdmissionWebhooks(ctx context.Context) error {
 	}
 
 	// Keep setting the caBundle in the expected webhook configurations.
-	go func() {
-		// Only inject if we've an explicit CA bundle ourselves. Otherwise the webhook configs
-		// may already have been created with one.
-		if len(caBundle) == 0 {
-			return
-		}
-		// Initial sleep for the client to initialize before our first calls.
-		// Ideally we could explicitly wait for it.
-		time.Sleep(5 * time.Second)
+	err = ctrl.NewControllerManagedBy(o.manager).
+		Named("validating-webhook-cert-updater").
```

Comment:
I believe you'll need a
```go
		// Filter events without changes for all watches.
		WithEventFilter(predicate.ResourceVersionChangedPredicate{}).
```
to prevent infinite reconciling, no? This is used in our other event-based controllers, e.g. [`operator_config.go`](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/21478f108665efe083f1ae90cea8f542d3b5044c/pkg/operator/collection.go#L79-L80).